apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PartDEventTransformer.java

```diff
@@ -245,27 +246,43 @@ private static ExplanationOfBenefit transformClaim(PartDEvent claimGroup) {
             TransformerUtils.createExtensionQuantity(
                 CcwCodebookVariable.DAYS_SUPLY_NUM, claimGroup.getDaysSupply()));
 
-    // TODO CBBD-241 - This code was commented out because values other than
-    // "01"
-    // were coming thru
-    // such as "07". Need to discuss with Karl if this check needs to be
-    // here -
-
     /*
-     * if (claimGroup.serviceProviderIdQualiferCode == null ||
-     * !claimGroup.serviceProviderIdQualiferCode.equalsIgnoreCase("01"))
-     * throw new InvalidRifValueException(
-     * "Service Provider ID Qualifier Code is invalid: " +
-     * claimGroup.serviceProviderIdQualiferCode);
+     *   Code	    Code value
+     *   01        National Provider Identifier (NPI)
+     *   06        Unique Physician Identification Number (UPIN)
+     *   07        National Council for Prescription Drug Programs (NCPDP) provider identifier
+     *   08        State license number
+     *   11        Federal tax number
+     *   99        Other
      */
 
+    String identiferSystem;
+
     if (!claimGroup.getServiceProviderId().isEmpty()) {
-      eob.setOrganization(
-          TransformerUtils.createIdentifierReference(
-              TransformerConstants.CODING_NPI_US, claimGroup.getServiceProviderId()));
-      eob.setFacility(
-          TransformerUtils.createIdentifierReference(
-              TransformerConstants.CODING_NPI_US, claimGroup.getServiceProviderId()));
+      switch (claimGroup.getServiceProviderIdQualiferCode()) {
+        case "01":
+          identiferSystem = TransformerConstants.CODING_NPI_US;
+          break;
+        case "06":
+        case "07":
+        case "08":
+        case "11":
+          identiferSystem = TransformerConstants.CODING_CODESYSTEM;
+          break;
+        default:
+          identiferSystem = null;
+          break;
+      }
+
+      if (StringUtils.isNotBlank(identiferSystem)) {
+        eob.setOrganization(
+            TransformerUtils.createIdentifierReference(
+                identiferSystem, claimGroup.getServiceProviderId()));
+        eob.setFacility(
+            TransformerUtils.createIdentifierReference(
+                identiferSystem, claimGroup.getServiceProviderId()));
+      }
```

Comment:
This won't quite work for our users. If you're going to use the <https://terminology.hl7.org/1.0.0/CodeSystem-v2-0203.html> codeset, then you have to use the codes from that codeset, too.

I'm having trouble figuring out how to type out a complete explanation, so perhaps we should chat about it on the phone next week? Dave may be able to help you with this, as well, if I get too tied up.